### PR TITLE
Add change log for upcoming release

### PR DESCRIPTION
**Pull request by trask**
Draft for now as there are still PRs to be merged: https://github.com/open-telemetry/opentelemetry-java-instrumentation/milestones/v1.26.0

### CODE REVIEW

**Comment by laurit** [comment]
@opentelemetrybot update

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,44 @@
...
+- Add instrumentation for vertx-sql-client
+  ([#8311](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/8311))
+
+### 🌟 New library instrumentation
+
+### 📈 Enhancements
+
+- Make net.transport an optional attribute
+  ([#8279](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/8279))
+- AddingSpanAttributes annotation
```
Comment (theletterf) [inline CHANGELOG.md:18]: Is that the name or there's a space missing between `Adding` and `SpanAttributes`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,44 @@
...
+  ([#7787](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/7787))
+- JavaScript Snippet Injection
```
Comment (theletterf) [inline CHANGELOG.md:20]: Does it need title capitalization?

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,44 @@
...
+  ([#7650](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/7650))
+- Updating Sockpeer service.
```
Comment (theletterf) [inline CHANGELOG.md:22]:
```suggestion
- Updating Sockpeer service
```

For formatting consistency.

File: CHANGELOG.md (hunk 1 above)
Comment (breedx-splk) [reply CHANGELOG.md:18]: That's the literal name of the attribute [@AddingSpanAttributes](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation-annotations/src/main/java/io/opentelemetry/instrumentation/annotations/AddingSpanAttributes.java#L35). It's a new feature that allows one to annotate a method to add attributes to existing parent context without creating a new span.

File: CHANGELOG.md (hunk 2 above)
Comment (breedx-splk) [reply CHANGELOG.md:20]: I don't think so...but that was the original case of the PR title.
```suggestion
- JavaScript snippet Injection
```

File: CHANGELOG.md (hunk 1 above)
Comment (trask) [reply CHANGELOG.md]: thx! I updated this to make it clearer

**Review by breedx-splk — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,47 @@
...
+
+### 🛠️ Bug fixes
+
+- Fix nested http.route
+  ([#8282](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/8282))
+- Fix undertow instrumentation with http pipelining
+  ([#8400](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/8400))
+- Fix http pipelining on Grizzly
+  ([#8411](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/8411))
+- Fix http pipelining Netty 4.1 server
```
Comment (mateuszrzeszutek) [inline CHANGELOG.md:38]:
```suggestion
- Fix http pipelining on Netty 4.1 server
```

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**